src/sleep/sleep.c

```diff
@@ -263,41 +263,91 @@ static int execute(
 }
 
 static int execute_s2h(const SleepConfig *sleep_config) {
-        _cleanup_close_ int tfd = -1;
-        struct itimerspec ts = {};
+        usec_t suspend_interval = sleep_config->hibernate_delay_sec, before_timestamp = 0, after_timestamp = 0;
+        bool woken_by_timer;
         int r;
 
         assert(sleep_config);
 
-        tfd = timerfd_create(CLOCK_BOOTTIME_ALARM, TFD_NONBLOCK|TFD_CLOEXEC);
-        if (tfd < 0)
-                return log_error_errno(errno, "Error creating timerfd: %m");
-
-        log_debug("Set timerfd wake alarm for %s",
-                  FORMAT_TIMESPAN(sleep_config->hibernate_delay_sec, USEC_PER_SEC));
+        while(!battery_is_low()) {
```

Comment:
Ahh, yes sure. Fixed.

Reply:
this while loop would run when battery is not low. battery_is_low is returning 1 when battery is low so here the condition should be `(battery_is_low() != 1)`

Reply:
> this while loop would run when battery is not low. battery_is_low is returning 1 when battery is low so here the condition should be `(battery_is_low() != 1)`

yeah, my bad. But please use `== 0`, or `<= 0`, instead of `!= 1`.